Makefile

```diff
@@ -12,22 +14,39 @@ help: ## Show help
 install: ## Install required dependencies
 	@if [ -z $(POETRY) ]; then \
 		echo "Poetry could not be found, installing..."; \
-		pip install poetry;\
+		$(PIP) install poetry; \
 	else \
-		poetry install;\
+		poetry install; \
 	fi
-	pip install pre-commit
+
+	$(PIP) install pre-commit;
 	pre-commit install
 
 remove: ## Remove poetry virtualenv
-	@echo "Removing virtualenv $(VIRTUALENV)"
-	poetry env remove $(VIRTUALENV)
+	@if [ -z $(POETRY) ]; then \
+  		echo $(POETRY_NOT_INSTALLED_MESSAGE); \
+	else \
+		@echo "Removing virtualenv $(VIRTUALENV)"; \
+		poetry env remove $(VIRTUALENV); \
+	fi
 
 serve: ## Serve mkdocs in local
-	poetry run mkdocs serve --strict -w src
+	@if [ -z $(POETRY) ]; then \
```

Comment:
it can happen, that poetry is installed, but virtualenv for the project is not
![image](https://github.com/McLargo/python-lessons/assets/690606/9404a1c7-eb40-4f2e-a50c-8635ea9d7c21)

lets check also if `VIRTUALENV` is in place

Reply:
additional checks added.